### DISCUSSION TITLE
How can a boundary condition of the form \frac{d u}{dx} = 0 = 0,  \frac{d u}{dy} = 0 , at   x,y = (0,0)?

### DISCUSSION THREAD

**Discussion by rafael-fuente** [lululxvi/deepxde#1901] [Q&A]
I'm trying to find solutions to a 2D PDE that satisfy that both derivatives are 0 at (x,y) = (0,0)

$\frac{d u}{dx} = 0 = 0,  \frac{d u}{dy} = 0$      at   the center of th domain   $(x,y) = (0,0)$

I've tried defining them as Neumann boundary conditions when solving Poisson equation $-u_{xx} - u_{yy} - 1 = 0$ in the domain $\Omega=[-1,1]^2$:
```python

from deepxde.backend import set_default_backend
set_default_backend('pytorch')

import deepxde as dde
import numpy as np
from deepxde.backend import torch


import deepxde as dde
import numpy as np


# Define the geometry
geom = dde.geometry.Rectangle([-1, -1], [1, 1])

# Define the PDE function
def pde(x, u):
    u_xx = dde.grad.hessian(u, x, i=0, j=0)
    u_yy = dde.grad.hessian(u, x, i=1, j=1)
    return -u_xx - u_yy - 1



bc = dde.icbc.NeumannBC(
    geom,
    lambda x: 0.0,
    lambda x, on_boundary: dde.utils.isclose(x[0], 0) and dde.utils.isclose(x[1], 0),
)


# Define the data object
data = dde.data.PDE(geom, pde, [bc], num_domain=2540, num_boundary=80, solution=None, num_test=1000)

# Define the neural network
net = dde.nn.FNN([2] + [20] * 3 + [1], "tanh", "Glorot uniform")

# Define the model
model = dde.Model(data, net)

# Compile the model
model.compile("adam", lr=0.001)

# Train the model
losshistory, train_state = model.train(epochs=5000)

# Plot the results
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

But I get the error: 
`ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 1 dimension(s)
`

Note that, however, If I change the boundary conditions to be satisfied along the entire line x = 0 instead of at a point (x,y) = (0,0), I don't get the error:

```python
bc = dde.icbc.NeumannBC(
    geom,
    lambda x: 0.0,
    lambda x, on_boundary: dde.utils.isclose(x[0], 0)
)
```

But I'm interested in the problem that they are only defined at  $(x,y) = (0,0)$

**Comment by lululxvi** [comment]
You need to guarantee [0, 0] is in the training points by using the anchor point.
